### PR TITLE
Refresh interop docs.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: INTEROP.md [hunk 1]
```diff
@@ -20,60 +20,51 @@ imported into an IDE for the purpose of code completion and navigation.
  Interoperability with Swift/Objective-C is provided too and covered in a
 separate document [OBJC_INTEROP.md](OBJC_INTEROP.md).
 
+## Platform libraries ##
+
+ Note that in many cases there's no need to use mechanisms described below, as for APIs available on the
+platform standartized bindings called platform libraries [PLATFORM_LIBS.md](PLATFORM_LIBS.md) could be used.
+For example, POSIX on Linux/macOS platforms, Win32 on Windows platform, or Apple frameworks on macOS/iOS are
+avilable this way.
```
Comment (sbogolepov) [inline INTEROP.md]: "available".

File: INTEROP.md [hunk 2]
```diff
@@ -20,60 +20,51 @@ imported into an IDE for the purpose of code completion and navigation.
  Interoperability with Swift/Objective-C is provided too and covered in a
 separate document [OBJC_INTEROP.md](OBJC_INTEROP.md).
 
+## Platform libraries ##
+
+ Note that in many cases there's no need to use mechanisms described below, as for APIs available on the
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: The entire document is "below", so this note is very confusing.

File: INTEROP.md [hunk 3]
```diff
@@ -20,60 +20,51 @@ imported into an IDE for the purpose of code completion and navigation.
...
+platform standartized bindings called platform libraries [PLATFORM_LIBS.md](PLATFORM_LIBS.md) could be used.
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: > `platform libraries [PLATFORM_LIBS.md](PLATFORM_LIBS.md)`

May be `[platform libraries](PLATFORM_LIBS.md)` ?

File: INTEROP.md [hunk 4]
```diff
@@ -20,60 +20,51 @@ imported into an IDE for the purpose of code completion and navigation.
...
+For example, POSIX on Linux/macOS platforms, Win32 on Windows platform, or Apple frameworks on macOS/iOS are
+avilable this way.
+
 ## Simple example ##
 
 Build the dependencies and compiler (see `README.md`).
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: Do we really need an example which is that hard to play with?

File: INTEROP.md [hunk 5]
```diff
@@ -636,9 +628,12 @@ The `.def` file supports several options for adjusting the generated bindings.
     values correspondingly. If the enum is not included into any of these lists,
     then it is generated according to the heuristics.
 
+*    `noStringConversion` property value is space-separated lists of the function whose
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: list of the functions ?

File: INTEROP.md [hunk 6]
```diff
@@ -672,3 +667,32 @@ fun zeroMemory(buffer: COpaquePointer, size: Int) {
 
 Also, the type parameter can be inferred automatically and so may be omitted
 in some cases.
+
+
+### Object pinning ###
+
+ Kotlin objects could be pinned, i.e. their position is memory is guaranteed to be stable
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: in memory

File: INTEROP.md (hunk 1 above)
Comment (olonho) [reply INTEROP.md]: Done.

File: INTEROP.md (hunk 2 above)
Comment (olonho) [reply INTEROP.md]: Done.

File: INTEROP.md (hunk 3 above)
Comment (olonho) [reply INTEROP.md]: Done.

File: INTEROP.md (hunk 4 above)
Comment (olonho) [reply INTEROP.md]: Done.

File: INTEROP.md (hunk 5 above)
Comment (olonho) [reply INTEROP.md]: Done.

File: INTEROP.md (hunk 6 above)
Comment (olonho) [reply INTEROP.md]: Done.

File: FAQ.md [hunk 7]
```diff
@@ -1,8 +1,9 @@
 ### Q: How do I run my program?
 
-A: Define a top level function `fun main(args: Array<String>)`, please ensure it's not
-in a package. Also compiler switch `-entry` could be used to make any function taking
-`Array<String>` and return `Unit` as an entry point.
+A: Define a top level function `fun main(args: Array<String>)` or just  `fun main()` if you not interested
```
Comment (SvyatoslavScherbina) [inline FAQ.md]: > you not

you are not?

File: INTEROP.md [hunk 8]
```diff
@@ -168,9 +168,38 @@ excludeDependentModules = true
 When both `excludeDependentModules` and `headerFilter` are used, they are
 applied as an intersection.
 
+### C compiler and linker options ###
+
+ Options passed to the C compiler (used to analyze headers, such as preprocessor definitions) and the linker
+(used to link final executables) can be passed in in the definition file as `compilerOpts` and `linkerOpts`
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: in in

File: INTEROP.md [hunk 9]
```diff
@@ -168,9 +168,38 @@ excludeDependentModules = true
...
+respectively. For example
+
+<div class="sample" markdown="1" theme="idea" mode="c">
+
+```c
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: This code is not written in C (the same applies for the code block below).

File: INTEROP.md [hunk 10]
```diff
@@ -168,9 +168,38 @@ excludeDependentModules = true
...
+compilerOpts = -DFOO=bar
+linkerOpts = -lpng
+```
+
+</div>
+
+Target-specific options, only applicable to the certain target can be specified as well, such as
```
Comment (SvyatoslavScherbina) [inline INTEROP.md]: It is not clear that any `.def` option can be target-specific, including e.g. `headers`. Is this miss intentional?

Comment (olonho) [reply INTEROP.md]: Nope, will fix.

File: INTEROP.md (hunk 8 above)
Comment (olonho) [reply INTEROP.md]: Done

**Pull request merged**